gendox-core-api/src/main/java/dev/ctrlspace/gendox/gendoxcoreapi/discord/ListenerService.java

```diff
@@ -204,4 +205,21 @@ public String getTheQuestion(SlashCommandInteractionEvent event) {
         return question;
     }
 
+    public boolean IsUserExist(String author) throws GendoxException{
+        return userService.isUserExistByUserName(author);
+    }
+
+    public User createDiscordUser(String author) throws GendoxException{
+        User user = new User();
+        user.setUserName(author);
+        user.setGlobalRole(typeService.getGlobalApplicationRoleTypeByName("ROLE_USER"));
+        user.setUserType(typeService.getUserTypeByName("DISCORD_USER"));
+
+
+        user = userService.createUser(user);
+
+        return user;
+
+    }
```

Comment:
This needs to go to `UserService`, Listener Service is not responsible to create users.